### DISCUSSION TITLE
Need help using ical4j urgently!!

### DISCUSSION THREAD

**Discussion by iliasworldpeace** [ical4j/ical4j#495] [Q&A]
Hello all,

First of all, some background: I'm a young programming student with little programming knowledge. I only know a bit of Python and that's it. For a school project I need to make an app where users can make and edit appointments. We need to do this with a program called Mendix, which you may or may not be familiar with. 

The appointment creation part is said and done and the only thing I need now is a feature that will let the user download their appointment in format of an ics file. I've looked and searched everywhere on how to do this but to no avail. Fortunately Mendix supports custom Java scripts.. I stumbled upon ical4j and it seems like exactly what I need, but like I said, I don't know the first thing about Java.

So if anyone is interested in helping a struggling student out a monetary reward would be in order (don't expect too much though because as I said I am a struggling student). If you want to help out of the kindness of your heart then that would be even better. I just need this done by wednesday (the 25th). If you need any more details just ask.

**Comment by benfortuna** [comment]
Hi Ilia,

Whilst I can't offer to help you directly I can provide a couple of pointers on where to start. 

First, if you need to integrate your program with Mendix you need to look at the languages it supports; if custom JavaScript is the only option then perhaps try a google search for "javascript icalendar". I think there are a couple of libraries but not sure.

Second, if you know Python and can do it in Python, there are also some good libraries available, eg. https://icalendar.readthedocs.io/en/latest/usage.html#example

Finally, if you do need to use Java, there are some examples of ical4j usage in the user guide. They are not very extensive/up-to-date but if you get stuck just post your issue here and will try to help:

https://ical4j.github.io/ical4j-user-guide/examples/

**Reply by iliasworldpeace** [reply]
Hey Ben, 

First of all, thanks for your reply. Mendix supports both java and javascript but not python as far as I know. I followed up on your advice and looked for some JavaScript libraries as js is much easier to implement in Mendix than Java. I found [ics.js](https://github.com/nwcell/ics.js) and I tried using it but I get an error saying ics is not defined. Like I said I don't know the first thing about JS. I installed npm and then used npm to install ics.js in my project directory. From there I thought I had to use the import function to import it into the code but that didn't seem to work for some reason so I ended up removing the import line.
This is the exact error: An error occurred while executing an action of MyFirstModule.Home_Web.actionButton2: ics is not defined 
Mind you, the actionButton2 mentioned is the button that I have assigned the javascript to.
Here is the code:
[source.txt](https://github.com/ical4j/ical4j/files/6532079/source.txt)
